jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -604,10 +604,14 @@ protected boolean passConditionalHeaders(HttpServletRequest request, HttpServlet
             }
 
             // Parse the if[un]modified dates and compare to resource
-            if (ifums != -1 && content.getResource().lastModified() / 1000 > ifums / 1000)
+            if (ifums != null && ifm == null)
```

Comment:
Since this is the `ResourceService` (handling only `GET` and `HEAD`), the `If-Unmodified-*` headers are never going to be seen here.

Reply:
Is it "never" or just very VERY rarely?
I'm inclined to leave support in for 10/11 and remove support in 12.

Reply:
Prior to RFC9110, the `If-Unmodified-Since` header was for unsafe methods (like `POST`) and had no meaning for safe methods (like `GET`).

But now https://www.rfc-editor.org/rfc/rfc9110#name-if-unmodified-since says ..

> A client MAY send an If-Unmodified-Since header field in a [GET](https://www.rfc-editor.org/rfc/rfc9110#GET) request to indicate that it would prefer a [412 (Precondition Failed)](https://www.rfc-editor.org/rfc/rfc9110#status.412) response if the selected representation has been modified. However, this is only useful in range requests ([Section 14](https://www.rfc-editor.org/rfc/rfc9110#range.requests)) for completing a previously received partial representation when there is no desire for a new representation. [If-Range](https://www.rfc-editor.org/rfc/rfc9110#field.if-range) ([Section 13.1.5](https://www.rfc-editor.org/rfc/rfc9110#field.if-range)) is better suited for range requests when the client prefers to receive a new representation.

So I'm actually thinking we should instead add back support in Jetty 12.